src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1785,6 +1785,11 @@ static async Task ExecuteAwaited(ValueTask task)
             await task;
         }
 
+        if (task.IsCompletedSuccessfully)
+        {
+            task.GetAwaiter().GetResult();
```

Comment:
btw there is a bug here. There should be a return after this line. Otherwise, it'll call `ExecuteAwaited` for a completed task anyway.

Reply:
(no real-world impact, but it's not working as intended)

Reply:
Discussed offline and we'll address this in the forward port to main.